src/core/cgroup.c

```diff
@@ -4186,7 +4200,10 @@ int unit_cgroup_freezer_action(Unit *u, FreezerAction action) {
 
         if (target == kernel) {
                 u->freezer_state = target;
-                return 0;
+                if (action == FREEZER_FREEZE)
+                        return 0;
+                else
+                        ret = 0;
         }
```

Comment:
to make this clearer, i wouldn't initialize ret = 1 up top, but simple here in an else branch.

i.e.:

```c
if (target == u->kernel) {
        u->freezer_state = target;
        if (action == FREEZER_FREEZE)
                return 0;
        ret = 0;
else
        ret = 1;
```

i think thta makes things a lot clearer